aea/cli/check_packages.py

```diff
@@ -258,6 +309,30 @@ def check_author(configuration_file: Path, expected_author: str) -> None:
         raise UnexpectedAuthorError(configuration_file, expected_author, actual_author)
 
 
+def check_public_id(configuration_file: Path) -> None:
+    """Check the public_id in the code and configuration match."""
+    expected_public_id = get_public_id_from_yaml(configuration_file)
+    # remove last 's' character (as package type is plural in packages directory)
+    package_type_str = configuration_file.parent.parent.name[:-1]
+    package_type = PackageType(package_type_str)
+    if package_type == PackageType.CONNECTION:
+        module_name_to_load = Path("connection.py")
+    elif package_type == PackageType.SKILL:
+        module_name_to_load = Path("__init__.py")
+    else:
+        # no check to do.
+        return
+    module_path_to_load = configuration_file.parent / module_name_to_load
+    content = module_path_to_load.read_text()
+    matches = re.findall("^PUBLIC_ID = (.*)", content, re.MULTILINE)
+    if len(matches) != 1:
+        raise PublicIdDefinitionError(package_type, expected_public_id, len(matches))
+
+    public_id_code = matches[0]
+    if str(expected_public_id) not in public_id_code:
+        raise WrongPublicIdError(package_type, expected_public_id, public_id_code)
```

Comment:
just saying: a false negative would be the string:

```python
PUBLIC_ID = PublicId("author", "name", "version")
```

Anyway, by convention we can say you have to use `PublicId.from_str`. 

Surely safer than using `eval` :)

Reply:
Yes I know - this is a bit hacky. I didn't feel like coming up with the comprehensive regex.

Reply:
Maybe you can generalise it in the next PR on this @marcofavorito ?

Reply:
Yes. we could cover the other most probable pattern (see above). Something like:
```
^PUBLIC_ID *= *PublicId\( *("[a-zA-Z0-9_]+") *, *("[a-zA-Z0-9_]+") *, *("[a-zA-Z0-9_]+") *\)
```

Reply:
Yes I think these are the two only ways of instantiating the object. And then we impose this as the requirement.